evm-validate/src/lib.rs

```diff
@@ -417,3 +422,331 @@ fn validate_transaction_hashes(
+        .context("get log block num col as u64")?;
+
+    let log_tx_idx = logs
+        .column_by_name("transaction_index")
+        .context("get tx index column")?
+        .as_any()
+        .downcast_ref::<UInt64Array>()
+        .context("get tx index col as u64")?;
+    
+    let log_address = logs
+        .column_by_name("address")
+        .context("get address column")?
+        .as_any()
+        .downcast_ref::<BinaryArray>()
+        .context("get address as binary")?;
+
+    let log_topic0 = logs
+        .column_by_name("topic0")
+        .context("get topic0 column")?
+        .as_any()
+        .downcast_ref::<BinaryArray>()
+        .context("get topic0 as binary")?;
+
+    let log_topic1 = logs
+        .column_by_name("topic1")
+        .context("get topic1 column")?
+        .as_any()
+        .downcast_ref::<BinaryArray>()
+        .context("get topic1 as binary")?;
+
+    let log_topic2 = logs
+        .column_by_name("topic2")
+        .context("get topic2 column")?
+        .as_any()
+        .downcast_ref::<BinaryArray>()
+        .context("get topic2 as binary")?;
+
+    let log_topic3 = logs
+        .column_by_name("topic3")
+        .context("get topic3 column")?
+        .as_any()
+        .downcast_ref::<BinaryArray>()
+        .context("get topic3 as binary")?;
+    
+    let log_data = logs
+        .column_by_name("data")
+        .context("get data column")?
+        .as_any()
+        .downcast_ref::<BinaryArray>()
+        .context("get data as binary")?;
+
+    // Return the extracted data
+    Ok((log_block_nums, log_tx_idx, log_address, log_topic0, log_topic1, log_topic2, log_topic3, log_data))
+}
+
+fn extract_transaction_cols_as_arrays(transactions: &RecordBatch) -> Result<(
+    &PrimitiveArray<UInt64Type>,
+    &PrimitiveArray<UInt64Type>,
+    &PrimitiveArray<UInt8Type>,
+    &GenericByteArray<GenericBinaryType<i32>>,
+    &GenericByteArray<GenericBinaryType<i32>>,
+    &PrimitiveArray<UInt8Type>,
+)> {
+    let tx_block_nums = transactions
+    .column_by_name("block_number")
+    .context("get tx block num col")?
+    .as_any()
+    .downcast_ref::<UInt64Array>()
+    .context("get tx block num col as u64")?;
+
+let tx_tx_idx = transactions
+    .column_by_name("transaction_index")
+    .context("get tx index column")?
+    .as_any()
+    .downcast_ref::<UInt64Array>()
+    .context("get tx index col as u64")?;
+
+let tx_status = transactions
+    .column_by_name("status")
+    .context("get tx status column")?
+    .as_any()
+    .downcast_ref::<UInt8Array>()
+    .context("get tx status col as u8")?;
+
+// Using cumulative_gas_used is a binary column, and converting to u64 is a hack
+// TODO: Find why I can't use the column directly as a u64
+let tx_cumulative_gas_used = transactions
+    .column_by_name("cumulative_gas_used")
+    .context("get tx cumulative gas used column")?
+    .as_any()
+    .downcast_ref::<BinaryArray>()
+    .context("get tx cumulative gas used col as binary")?;
```

Comment:
I should be able to cast `cumulative_gas_used` as `UInt64Array`. It's not a U256 value. I need to figure out why.